File: archive/divide-array-into-equal-pairs.js
const divideArray = function (nums) {
  let mem = {}
  nums.forEach(el => {
    if (mem[el] === undefined) {
      mem[el] = 1
    } else {
      mem[el]++
    }
  })

  let ans = true
  Object.keys(mem).forEach(k => {
    if (mem[k] % 2 !== 0) {
      ans = false
    }
  })

  return ans
}

console.log(divideArray([3, 2, 3, 2, 2, 2]))

File: go-archive/divide-array-into-equal-pairs.go
func divideArray(nums []int) bool {
    mem:=map[int]int{}

    for i:=0; i < len(nums); i++ {
        if _, isHas := mem[nums[i]]; isHas {
            mem[nums[i]]++
        } else {
            mem[nums[i]] = 1
        }
    }

    for _, v := range mem {
        if (v % 2 == 1) {return false}
    }

    return true
}